src/core/types/addons.js

```diff
@@ -53,7 +53,7 @@ type PartialExternalAddonVersionType = {|
 export type ExternalAddonVersionType = {|
   ...PartialExternalAddonVersionType,
   // The `text` property is omitted from addon.current_version.license.
-  license: { name: string, url: string },
+  license?: { name: string, url: string },
```

Comment:
The type you changed here is supposed to have a defined `license`. The one without a license is [`PartialExternalAddonVersionType`](https://github.com/mozilla/addons-frontend/blob/4181a0940045b271fae07fec217f70a9da61709e/src/core/types/addons.js#L39). Sometimes it's hard to make this kind of thing work with Flow but could you check the place where you expected to get a warning about an undefined `license`? You might be able to fix it by declaring the input as `PartialExternalAddonVersionType | ExternalAddonVersionType`

Reply:
Ah, that makes more sense. I wasn't sure what the heck `PartialExternalAddonVersionType` was even for! I wasn't actually getting any warnings without the `?`, but I knew from the code I had to change that when recommendations are returned we don't get a `license` or `release_notes` for the `current_version`.

If I remove the `?` Flow is happy, but I think maybe something is incorrect somewhere. I tried to use `PartialExternalAddonVersionType` but then Flow had me running in circles because suddenly there were all kinds of errors.

@kumar303 Do you think this is something that needs to be fixed, or is it okay to just leave it as is, without the `?`.

Reply:
> I wasn't sure what the heck PartialExternalAddonVersionType was even for!

Maybe add a comment above it that would help your former self understand it?

> is it okay to just leave it as is, without the `?`.

Yeah, I suggest doing that. It's a bit of a headache trying to get Flow to understand union types consisting of complex objects so no need to spend a lot of time on it.

Reply:
Nevermind, I got it working, although I'm not sure if it's correct. I'll flag you for a review, @kumar303.